provisioner/shell/provisioner.go

```diff
@@ -201,6 +199,11 @@ func (p *Provisioner) Provision(ctx context.Context, ui packer.Ui, comm packer.C
 		writer := bufio.NewWriter(tf)
 		writer.WriteString(fmt.Sprintf("#!%s\n", p.config.InlineShebang))
 		for _, command := range p.config.Inline {
+			p.config.ctx.Data = generatedData
```

Comment:
Why not have this up by line 182 so we don't repeatedly re-set it for each line in inline?